### PR TITLE
`useImperativeHandle` hook is refactored

### PR DESCRIPTION
**Pull request by sgrishchenko**
cc @turansky @aerialist7

### CODE REVIEW

File: examples/src/main/kotlin/example/ForwardingRefs.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    useImperativeHandle(forwardedRef, inputRef) {
+        object : Focusable {
+            override fun focus() {
+                inputRef.current?.focus()
+            }
+        }
+    }
+
+    input {
+        attrs {
```
Comment (turansky) [inline examples/src/main/kotlin/example/ForwardingRefs.kt:23]: `attrs` can be removed
`react.dom.html.ref` - Strict ref

File: examples/src/main/kotlin/example/ForwardingRefs.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+external interface Focusable {
+    fun focus()
+}
+
+val FancyInput = forwardRef<Props> { _, forwardedRef ->
+    val inputRef = useRef<HTMLInputElement>()
+
+    useImperativeHandle(forwardedRef, inputRef) {
+        object : Focusable {
```
Comment (turansky) [inline examples/src/main/kotlin/example/ForwardingRefs.kt:15]: `jso`?
Do you really need additional class?

**Review by turansky — CHANGES_REQUESTED** [review]

File: examples/src/main/kotlin/example/ForwardingRefs.kt [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+val App = fc<Props> {
+    val fancyInputRef = useRef<Focusable>()
+
+    useEffectOnce {
+        fancyInputRef.current?.focus()
+    }
+
+    FancyInput {
+        attrs {
+            ref = fancyInputRef
```
Comment (turansky) [inline examples/src/main/kotlin/example/ForwardingRefs.kt:38]: Request discussion about "forward ref component"

**Review by turansky — CHANGES_REQUESTED** [review]

File: examples/src/main/kotlin/example/ForwardingRefs.kt (hunk 1 above)
Comment (sgrishchenko) [reply examples/src/main/kotlin/example/ForwardingRefs.kt:23]: in `React` ref is regular prop, I don't know why it was moved to `RElementBuilder`

File: examples/src/main/kotlin/example/ForwardingRefs.kt (hunk 2 above)
Comment (sgrishchenko) [reply examples/src/main/kotlin/example/ForwardingRefs.kt:15]: yes I do, I want to strictly implement external contract, `jso` is unsafe for it

File: examples/src/main/kotlin/example/ForwardingRefs.kt (hunk 3 above)
Comment (sgrishchenko) [reply examples/src/main/kotlin/example/ForwardingRefs.kt:38]: All in all, I think we can merge this PR

**Pull request merged**